eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/actions/OptionalAction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.actions;
+
+import com.schedjoules.client.eventsdiscovery.Event;
+import com.schedjoules.eventdiscovery.framework.utils.AbstractCachingOptional;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+import org.dmfs.httpessentials.types.Link;
+import org.dmfs.iterators.AbstractFilteredIterator;
+import org.dmfs.iterators.FilteredIterator;
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+import java.util.Iterator;
+
+
+/**
+ * {@link Optional<Action>} that looks for the {@link Action} in a provided {@link Iterable<Action>} by link rel type.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class OptionalAction extends AbstractCachingOptional<Action>
+{
+
+    public OptionalAction(final String actionLinkRelType, final Iterable<Link> actionLinks, final ActionFactory actionFactory, final Event event)
```

Comment:
There was an `Actions` `Iterable` earlier that created the `Action`s, but I needed to filter them by rel type and instead of adding that as method to `Action` (which I did at one point), I thought it may be better to just filter the `Link`s first and then create that single `Action`. So `OptionalAction` needs the factory to create the `Action` from the corresponding `Link`.